CMakeLists.txt

```diff
@@ -103,6 +105,25 @@ foreach(varName HAS_STD_LOCALE  HAS_ODBC  HAS_MBSRTOWCS  HAS_WCSTOMBS  HAS_FWIDE
   endif()
 endforeach()
 
+#
+# Package and sign if Apache maintainer
+#
+option(APACHE_MAINTAINER "Apache maintainer" OFF)
+if(APACHE_MAINTAINER)
+set(CPACK_SOURCE_PACKAGE_FILE_NAME "apache-log4cxx-${LOG4CXX_RELEASE_VERSION}")
+set(CPACK_SOURCE_GENERATOR "TGZ;ZIP")
+set(CPACK_SOURCE_IGNORE_FILES ".git/;build/")
+include(CPack)
+
+add_custom_target( dist 
+    COMMAND ${CMAKE_COMMAND} --build ${CMAKE_BINARY_DIR} -- package_source
+    COMMAND ${CMAKE_COMMAND} -E sha512sum "apache-log4cxx-${LOG4CXX_RELEASE_VERSION}.tar.gz" > "apache-log4cxx-${LOG4CXX_RELEASE_VERSION}.tar.gz.sha512"
```

Comment:
`sha512sum` is simply a binary expected to be available using `PATH`, correct? Because in my case it isn't, but the build seems to succeed. I didn't see any error message that this command or that for `gpg` failed. Though, only the archives themself got created.

Is that something to ignore or make more robust?

Reply:
The 'sha512sum' is actually a built-in CMake command: https://cmake.org/cmake/help/latest/manual/cmake.1.html#run-a-command-line-tool

As for the `gpg`, since that's really only needed for us as Apache maintainers, we could either ignore it, or we could make it somewhat more robust and have a separate 'sign' target.  This just makes it very easy to do it all at once.

Reply:
I was looking in the wrong directory, `_CPack_Packages` only. But the archives from there get copied into `out\build\x64-Debug` in the end and the SHA-files were properly there.

No need to to make things additionally complicated for `gpg` right now. In the worst case, missing signatures will most likely be spotted anyway.